tests/gui/components/drive-selector.spec.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+const _ = require('lodash');
+const m = require('mochainon');
+const angular = require('angular');
+require('angular-mocks');
+
+describe('Browser: DriveSelector', function() {
+
+  beforeEach(angular.mock.module(
+    require('../../../lib/gui/components/drive-selector/drive-selector')
+  ));
+
+  describe('DriveSelectorController', function() {
+
+    const defaultDrive = {
+      device: '/dev/disk2',
+      name: 'My Drive',
+      size: 123456789,
+      protected: false,
+      system: false
+    };
+
+    let currentDrive;
+
+    let $controller;
+    let $q;
+    let $rootScope;
+
+    const importSelectionStateModel = () => {
+      beforeEach(angular.mock.module({
+        SelectionStateModel: {
+          getDrive: () => {
+            return currentDrive;
+          },
+          hasDrive: () => {
+            return Boolean(currentDrive);
+          },
+          isSystemDrive: () => {
+            return Boolean(currentDrive.system);
+          }
+        }
+      }));
+    };
+
+    const injectDependencies = () => {
+      beforeEach(angular.mock.inject(function(_$controller_, _$q_, _$rootScope_) {
+        $controller = _$controller_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+      }));
+    };
+
+    // Note '$rootScope' won't be instantiated until inside an 'it',
+    // therefore only call this function there or in a deeper scope.
+    const makeDriveSelectorController = () => {
+      const scope = $rootScope.$new();
+
+      return $controller('DriveSelectorController', {
+        $scope: scope,
+
+        // Inject stub for $uibModalInstance; this is necessary because
+        // none is provided by UI Bootstrap.
+        $uibModalInstance: {
+          result: $q.resolve(),
+          opened: $q.resolve(),
+          closed: $q.resolve(),
+          rendered: $q.resolve(),
+          close: _.constant($q.resolve()),
+          dismiss: _.constant($q.reject())
+        }
+      });
+    };
+
+    describe('.getButtonStatus()', function() {
+
+      importSelectionStateModel();
+      injectDependencies();
+
+      describe('given there is no selected drive', function() {
+
+        it('should return "disabled"', function() {
+
+          const DriveSelectorController = makeDriveSelectorController();
```

Comment:
Instead of creating the controller on each test, you can do it on a `beforeEach` block, which will automatically reset the controller from scratch on each test.